lib/stupidedi/guides/004010/guide_builder.rb

```diff
@@ -50,9 +50,10 @@ def Element(requirement, name, *constraints)
 
         # @return [SegmentUse]
         def Segment(position, segment_def, name, requirement, repeat_count, *elements)
-          unless elements.length == segment_def.element_uses.length
+          num_required_elements = segment_def.element_uses.count{|x| x.required? }
```

Comment:
🍨

Reply:
This code is not the best. Really we should be validating that all the required fields are present, not that there are at least the same number of fields as there are required fields.